### PR TITLE
Change positioning engine from Tether to Popper.js

### PR DESCRIPTION
**Pull request by lancegliser**
The existing tether project has been discontinued as noted on their page. Many projects have begun to move away from tether, to Popper as suggested by Hubspot. My own need was because of Bootstrap 4's migration. I wanted Shepherd, but didn't want to include Tether just for it.

I expect this will have some bugs and use cases missing, but it's a functional start to begin talking about how options, themes, etc might change in a 2.x popper branch.

Conceptually, little has changed:
- The arrow css depends on a dom element instead of a :before as popper targets it differently

Outstanding issues:
- Orphan steps using positionFixed in popper's upcoming release

### CODE REVIEW

**Comment by lancegliser** [comment]
Associating this with #163

File: bower.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "tether-shepherd",
+  "name": "tether-popper",
```
Comment (rwwagner90) [inline bower.json]: Shouldn't this be `popper-shepherd`?

File: dist/css/shepherd-theme-arrows-fix.css [hunk 2]
```diff
@@ -1,5 +0,0 @@
-.shepherd-element-attached-bottom.shepherd-element-attached-right.shepherd-target-attached-top.shepherd-target-attached-left .shepherd-content:before,
```
Comment (rwwagner90) [inline dist/css/shepherd-theme-arrows-fix.css]: Is the arrow fix no longer needed?

**Comment by rwwagner90** [comment]
@lancegliser I tried this code out and it seems to work pretty well. How should we implement the orphaned steps stuff? Is that supported by popper now?

**Comment by rwwagner90** [comment]
I'm going to merge this into the 2.0-alpha branch for now, until I can get the orphaned elements to work.

**Comment by lancegliser** [comment]
@rwwagner90 Sorry I've been very bogged down with things. Happy to see this was a pleasing start to 2.x for you. I should have some time in the next month to attempt the orphan code as well if that's soon enough for you.

**Comment by rwwagner90** [comment]
@lancegliser I took a stab at it, but was unable to get offsets to work, so the orphaned element is not in the center of the screen. It may be easier to just copy the centered fixed logic from tether and not use popper for this case.

**Pull request merged**